### PR TITLE
Drop Python2 support and remove code supporting it

### PR DESCRIPTION
**Pull request by vrusinov**
Makes code cleaner and will allow using more modern language features

### CODE REVIEW

**Review by coddingtonbear — APPROVED** [review]
Oh, this is great, @vrusinov!  I'm going to hold off on merging this until I can get a :+1: from one of the other maintainers, though.

Have a few minutes to give this a glance, @ryneeverett?

**Review by ryneeverett — APPROVED** [review]
Diff looks good to me!

**Pull request merged**
